src/cryptoadvance/specter/managers/node_manager.py

```diff
@@ -0,0 +1,189 @@
+                user="",
+                password="",
+                port=8332,
+                host="localhost",
+                protocol="http",
+                external_node=True,
+                default_alias="default",
+            )
+        else:
+            self.nodes = nodes
+
+    @property
+    def active_node(self):
+        return self.get_by_alias(self._active_node)
+
+    @property
+    def nodes_names(self):
+        return sorted(self.nodes.keys())
+
+    def switch_node(self, node_alias):
+        # this will throw an error if the node doesn't exist
+        self._active_node = self.get_by_alias(node_alias).alias
+
+    def get_by_alias(self, alias):
+        for node_name in self.nodes:
+            if self.nodes[node_name] and self.nodes[node_name].alias == alias:
+                return self.nodes[node_name]
+        raise SpecterError("Node %s does not exist!" % alias)
+
+    def add_node(
+        self,
+        name,
+        autodetect,
+        datadir,
+        user,
+        password,
+        port,
+        host,
+        protocol,
+        external_node,
+        default_alias=None,
+    ):
+        if not default_alias:
+            node_alias = alias(name)
+        else:
+            node_alias = default_alias
+        fullpath = os.path.join(self.data_folder, "%s.json" % node_alias)
+        i = 2
+        while os.path.isfile(fullpath):
+            node_alias = alias("%s %d" % (name, i))
+            fullpath = os.path.join(self.data_folder, "%s.json" % node_alias)
+            i += 1
+
+        node = Node(
+            name,
+            node_alias,
+            autodetect,
+            datadir,
+            user,
+            password,
+            port,
+            host,
+            protocol,
+            external_node,
+            fullpath,
+            self,
+        )
+        write_node(node, fullpath)
+        self.update()  # reload files
+        return node
+
+    def add_internal_node(
+        self,
+        name,
+        default_alias=None,
+    ):
+        if not default_alias:
+            node_alias = alias(name)
+        else:
+            node_alias = default_alias
+        fullpath = os.path.join(self.data_folder, "%s.json" % node_alias)
+        i = 2
+        while os.path.isfile(fullpath):
+            node_alias = alias("%s %d" % (name, i))
+            fullpath = os.path.join(self.data_folder, "%s.json" % node_alias)
+            i += 1
+
+        node = InternalNode(
+            name,
+            node_alias,
+            False,
+            os.path.join(self.data_folder, f"{node_alias}/.bitcoin"),
+            "bitcoin",
+            secrets.token_urlsafe(16),
+            8332,
+            "localhost",
+            "http",
+            fullpath,
+            self,
+            self.bitcoind_path,
+            "mainnet",
+            self.internal_bitcoind_version,
+        )
+        write_node(node, fullpath)
+        self.update()  # reload files
+        return node
```

Comment:
Here as well. Always good to have some info in the logs if persistence data is changing.